packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -738,6 +759,13 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         }
     }
 
+    // Verify not tombstoned is seperate from not closed as the setConnectionState method is internal
+    private verifyNotTombstoned() {
+        if (this.tombstoned) {
+            throw new Error("Context is tombstoned!");
```

Comment:
I do think DataCorruptionError would strike the right tone here. We should maybe revisit what the recovery plan actually is.  And also if we want the "Escape valve" of only crashing this way once per X days.

Reply:
Do we need to implement the escape valve now? I thought we would implement it later

Reply:
I think we should consider using `SweepReadyUsageDetectionHandler.usageDetectedInInteractiveClient` to get the escape value going.  We can discuss what the right errorType should be, if it should change.

Reply:
May put this off, tried to figure it out without being hacky, but I didn't want to make that many changes to the datastorecontext. Might do it in another PR

Reply:
+1 on using data corruption. Let's not add the escape valve here just yet. After my work of adding tombstones to the summary is complete, we will have a clearer picture of what the overall flow looks like and we can add it then as needed.

Reply:
Not DataProcessing? DataCorruption

Reply:
Yes Corruption